www/js/metrics/MetricsDateSelect.tsx

```diff
@@ -0,0 +1,71 @@
+/* This button launches a modal to select a date range, which determines what time period
+    for which metrics should be displayed.
+  The button itself is a NavBarButton, which shows the currently selected date range,
+    a calendar icon, and launches the modal when clicked.
+  The modal is a DatePickerModal from react-native-paper-dates, which shows a calendar
+    and allows the user to select a date.
+*/
+
+import React, { useState, useCallback, useMemo } from "react";
+import { Text, StyleSheet } from "react-native";
+import { DatePickerModal } from "react-native-paper-dates";
+import { Divider, useTheme } from "react-native-paper";
+import i18next from "i18next";
+import { useTranslation } from "react-i18next";
+import NavBarButton from "../components/NavBarButton";
+import { DateTime } from "luxon";
+
+type Props = {
+  dateRange: DateTime[],
```

Comment:
we should set this to the first date of the study/program or the first date for the current user, similar to the "Download JSON dump" code in the profile.